api/app/models/comment.rb

```diff
@@ -28,9 +29,63 @@ class Comment < ApplicationRecord
                       counter_cache: :children_count
   has_many :children, class_name: "Comment", foreign_key: :parent_id, dependent: :destroy,
                       inverse_of: :parent, counter_cache: :children_count
-  has_many :flags, as: :flaggable, dependent: :destroy, inverse_of: :flaggable
+  has_one :comment_created_event, -> { where event_type: Event::COMMENT_CREATED },
+          class_name: Event,
+          as: :subject,
+          dependent: :destroy,
+          inverse_of: :subject
+  delegate :project, to: :subject
 
   # Validations
   validates :body, :subject, presence: true
 
+  # Callbacks
+  after_create :notify_parent_creator!, :notify_project_editors!
+  after_commit :trigger_event_creation, on: [:create]
+
+  # rubocop:disable Metrics/LineLength, Metrics/AbcSize
+  def subject_url
```

Comment:
I don't love this here, but it's OK for now, as I think we do something similar in other models. At some point, we should move this URL generation into a helper class of some sort.